### ISSUE TITLE
Dozens of chromium instances + very high memory usage

### ISSUE THREAD

**Issue by KevinNovak** [jef/streetmerchant#1930]
## Current Behavior

I'm running street merchant on my personal machine and expericing very high memory usage.

In task manager I see dozens of chromium instances, totaling to about 10 GB of memory usage:

![image](https://user-images.githubusercontent.com/11231402/107582634-c30cec00-6bc7-11eb-865a-8442dfd09377.png)

Is this the normal behavior?

## Steps to Reproduce

I'm using the following filters:
```txt
SHOW_ONLY_SERIES=3080
STORES=adorama,amazon,amd,bandh,bestbuy,corsair,evga,gamestop,microcenter,newegg,officedepot,pny,target,walmart,zotac
```

## Environment

- OS: Windows 10

**Comment by Henni16** [comment]
I have the same issue under MacOs. I tried to fix the code itself, unfortunately unsuccessfully so. I'm currently using the following dirty fix:
```#!/bin/bash
while true
do
npm run start
sleep 600
killall node
sleep 2
done

**Comment by mspych** [comment]
Same here. This is a real stress test for the M1 Mac 😄

**Comment by cabe1214** [comment]
> Is this the normal behavior or is there any way to limit the usage?
Here is my guess: YES, you will probably find one chromium process per store per URL.
So if **adorama** store has 10 different URLs for the 3080 (one per model) you'll see 10 chromium processes running.
If you have 10 stores selected and each store has 10 different URLs for the 3080, you'll end up running up to 100 process at the same time (probably not quite true as they won't run all in parallel, but you get me right?).

My suggestion: reduce the numbre of stores or filter just searching for some models.

**Comment by Henni16** [comment]
Unfortunately this does not solve the problem. I was only searching for the ps5 in about 7-8 different stores. I looked it up and there was a single URL for each store. I was still facing the problem described above.
Additionally, my fix of "restart the script every 10 minutes" wouldn't work if the problem was a high number of URLs that the script has to crawl. What seems to be happening is that some chrome instances aren't closed properly. After about 1 hour of usage it is still running fine and there are a couple of GB of RAM allocated. However, if I simply let the script do its thing for 4-5 hours my machine needs 8GB of swap memory in addition to the 16GB of RAM.

**Comment by cabe1214** [comment]
Hi @Henni16, did you have a look at #1801? 
I just saw that there are some problems with Mediamarkt/Saturn stores? IDK if they are among the stores you've configured.

**Comment by Henni16** [comment]
Ah nice thank you. Those two were indeed among the stores I configured.

**Comment by cabe1214** [comment]
You can give it a try and post here the result.
@KevinNovak if this is also your case (having Mediamarkt/Saturn among your stores) I think @jef could close this issue as duplicate of #1801

**Comment by Henni16** [comment]
I tested it for the past hour and the issue seems to be fixed. The memory usage now alternates between 500MB and 1.5GB, which is expected.

**Comment by KevinNovak** [comment]
@cabe1214 I don't have Mediamarkt/Saturn as one my stores.

Here are the stores I'm using:

* adorama
* amazon
* amd
* bandh
* bestbuy
* corsair
* evga
* gamestop
* microcenter
* newegg
* officedepot
* pny
* target
* walmart
* zotac

**Comment by cabe1214** [comment]
Hi @KevinNovak, I've counted +90 different URLs among all these stores for a 3080 card, so no doubt you end up having +50 chromium processes. In your case I think this behavior might be as expected.

**Comment by jef** [comment]
These chromium instances are not only the browser, but each tab.

We shouldn't be getting stale tabs, but this could be the case.
